### PR TITLE
Fix mypy for cli package

### PR DESCRIPTION
**Pull request by potiuk**
Part of #19891

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
cc: @josh-fell

**Comment by potiuk** [comment]
The CLI Mypy is also much nicer now!

**Comment by potiuk** [comment]
sounds like ready to go

File: airflow/cli/commands/cheat_sheet_command.py [hunk 1]
```diff
@@ -38,7 +37,13 @@ def display_recursive(
...
         groups: List[GroupCommand]
-        actions_iter, groups_iter = partition(lambda x: isinstance(x, GroupCommand), commands)
+        actions_iter: List[ActionCommand] = []
+        groups_iter: List[GroupCommand] = []
+        for command in commands:
+            if isinstance(command, GroupCommand):
+                groups_iter.append(command)
+            else:
+                actions_iter.append(command)
         actions, groups = list(actions_iter), list(groups_iter)
```
Comment (uranusjr) [inline airflow/cli/commands/cheat_sheet_command.py]:
```suggestion
        actions: List[ActionCommand] = []
        groups: List[GroupCommand] = []
        for command in commands:
            if isinstance(command, GroupCommand):
                groups.append(command)
            else:
                actions.append(command)
```

**Comment by uranusjr** [comment]
The entire anonymizer architecture should be added types as well. That can be done in another PR.

**Comment by potiuk** [comment]
Updated

**Review by uranusjr — APPROVED** [review]

**Pull request merged**